sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -115,6 +115,8 @@ object PhysicalOperation extends OperationHelper {
     val (fields, filters, child, _) = collectProjectsAndFilters(plan, alwaysInline)
     // If more than 2 filters are collected, they must all be deterministic.
     if (filters.length > 1) assert(filters.forall(_.deterministic))
+    // PythonUDFs should not be pushed
+    assert(filters.forall(!_.exists(_.isInstanceOf[PythonUDF])))
```

Comment:
shall we add the same assert in `ScanOperation`?

Reply:
Yeah